docs/DeveloperGuide.md

```diff
@@ -1484,17 +1474,30 @@ Assumptions: The sample data provided by InternBuddy is used, where there is a t
 
 <div style="page-break-after: always;"></div>
 
-## **Appendix C: Proposed Design Tweaks for Feature Flaws**
+## **Appendix C: Planned Enhancements**
 While we strive to make InternBuddy a perfect product for you, there are nevertheless areas of improvement.
 Hence, we are constantly striving to make InternBuddy better. This appendix documents some of the areas
 that our team is currently working on, and we would release them in the future once their functionalities
 have been finalised.
 
 ### Title of Design Tweak
 **Problem:**
-Explain the problem here
+Prefixes are case-sensitive, hence command arguments such as ` T/` will be interpreted as plain text. For example, `add n/Visa r/Software Engineer s/New d/2023-03-01 t/Golang T/c++` will add an internship entry with company name
+   `Visa`, role `Software Engineer`, status `New`, deadline of application `2023-03-01`, 
+   and tag `Golang T/c++`.
+  Therefore, it is possible to add substrings such as `T/`, `C/` or `R/` to any of the fields, even though the user could have intended to enter `t/`, `c/` or `r/`. 
+
+Moreoever, commmands such as `find` and `delete-field` do not consider prefixes such as `c/`, although other commands such as `edit` and `add` do  use the prefix `c/`. In the case of `find` and `delete-field`, it is possible to add the substring `c/` to any of the fields, even though the user could have intended to enter `c/` as a command prefix.
+
+We originally intended for this design to allow maximum flexibility for the user. If the user had entered something wrongly, it is possible to just use the command `edit`. However, this design could have an unintentional side-effect.
+
+ For example, using the `find` command, `find t/numbers t/cheese` tries to find entries with either the tag `numbers` or `cheese`. Tag matching is case-insensitive, so it also tries to find `Numbers` or `CheEse`. Similarly, `delete-field t/numbers t/cheese` tries to delete entries with either the tag `numbers` or `cheese` or `Numbers` or `CheEse`
+
+There could be another conflicting meaning of the command. In this case, `find t/numbers t/cheese` tries to find "numbers" or "cheese". but the user could have intended to find the tag `numbers T/cheese`. This could lead to some confusion.
+
+**Proposed Design Tweak**: Make prefixes case-insensitive. For example, `find t/numbers t/cheese` should have the same result as `find T/numbers T/cheese`. This would remove the confusion as substrings such as `T/` or `c/` cannot be entered in any of the fields. 
 
-**Proposed Design Tweak**: Explain the design tweak here
+Alternatively, we propose to implement a warning if strings such as ` T/` is detected in the user input. It is unlikely user intended to add `numbers T/cheese` as a tag and if the user added such a tag by mistake, the remedy is for the user to edit it.
```

Comment:
Would it be difficult to provide a bit more concrete implementation on how to make prefixes case-insensitive and/or how to implement the warning?

For example, which part of the code needs adjustment and/or what kind of warning message would be displayed to the user.

If it's too troublesome then it's okay.